### PR TITLE
MA-10: Create products page

### PR DESCRIPTION
**Pull request by vitalii-petrun**
https://user-images.githubusercontent.com/87064627/197469054-7aa7edbc-09ce-4453-9b8d-3ab1477f82f5.mp4

### CODE REVIEW

File: lib/presentation/pages/products_page/sections/gradient_section.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        children: [
+          const _InfoBlock(title: "We are trusted"),
+          ..._statsInfoBlocks(context),
+        ],
+      ),
+    );
+  }
+}
+
+List<_InfoBlock> _statsInfoBlocks(BuildContext context) {
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/gradient_section.dart:28]: Make List<Widget>

File: lib/presentation/pages/products_page/sections/partners_section.dart [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+    );
+  }
+}
+
+abstract class _UISettings {
+  static const double contentMaxWidth = 1270;
+  static const double smallScreenWidth = 780;
+  static const double horizontalPadding = 16;
+  static const double partnerCardMaxHeight = 470;
+}
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/partners_section.dart:42]: Pls delete this for now

File: lib/presentation/themes/src/theme_text_styles.dart [hunk 3]
```diff
@@ -50,6 +50,22 @@ class ThemeTextStyles extends ThemeExtension<ThemeTextStyles> {
   final TextStyle partnerTableSectionReferralCell;
   final TextStyle partnerPdfSectionTitle;
   final TextStyle partnerPdfSectionSubtitle;
+  final TextStyle productsHeadlineMedium;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_text_styles.dart:53]: Rename pls, delete "medium"

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/presentation/pages/products_page/sections/gradient_section.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
+part of '../products_page.dart';
+
+class _GradientSection extends StatelessWidget {
+  const _GradientSection();
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: context.screenWidth,
```
[the remark this answers is not in the document]
Comment (vitalii-petrun) [reply lib/presentation/pages/products_page/sections/gradient_section.dart:9]: it's not needed, sorry

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  _FaqItem(
+    text: "When can I test MultiMeta's metaverse product?",
+  ),
+  _FaqItem(
+    text:
+        "Where can I buy a MultiMeta Universe-compatible headset for traveling through the Metaverse?",
+  ),
+];
+
+class _FaqItem extends StatefulWidget {
```
[the remark this answers is not in the document]
Comment (vitalii-petrun) [reply lib/presentation/pages/products_page/sections/faq_section.dart:42]: my bad

File: lib/presentation/pages/products_page/products_page.dart [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        slivers: [
+          SliverPersistentHeader(
+            key: _headerKey,
+            delegate: Header(
+              headerYOffset: _getHeaderYOffset,
+              screenWidth: context.screenWidth,
+            ),
+            pinned: true,
+          ),
+          const SliverSizedBox(child: Align(child: _PresentationSection())),
```
[the remark this answers is not in the document]
Comment (vitalii-petrun) [reply lib/presentation/pages/products_page/products_page.dart:47]: to make constraints work

File: lib/presentation/pages/products_page/sections/investors_section.dart [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+class _InvestorsSection extends StatelessWidget {
+  const _InvestorsSection();
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      constraints: const BoxConstraints(maxWidth: 1270),
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: LayoutBuilder(
```
[the remark this answers is not in the document]
Comment (vitalii-petrun) [reply lib/presentation/pages/products_page/sections/investors_section.dart:11]: actually we can use Wrap, but on original web site this section has behaviour more like I described via LayoutBuilder

I think I shouls use just Wrap

File: lib/presentation/pages/products_page/sections/investors_section.dart [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        children: [
+          Text(
+            title,
+            style: context.text.productsInvestorSectionTitle,
+          ),
+          const SizedBox(height: 16),
+          SizedBox(
+            width: context.screenWidth < 780
+                ? context.screenWidth * 0.66
+                : context.screenWidth * 0.22,
```
[the remark this answers is not in the document]
Comment (vitalii-petrun) [reply lib/presentation/pages/products_page/sections/investors_section.dart:78]: I applied it for all InvestorComponent, as it was meant

it gives constrained and responsive blocks called InvestorComponent which are columns with text

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+class _FaqSection extends StatelessWidget {
+  const _FaqSection();
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      constraints: const BoxConstraints(maxWidth: 820),
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: Column(
```
[the remark this answers is not in the document]
Comment (vitalii-petrun) [reply lib/presentation/pages/products_page/sections/faq_section.dart:11]: we already have customscrollview, it's not recommended to use another scrollable (ListView.separated) inside of it

File: lib/l10n/app_ru.arb [hunk 10]
```diff
@@ -66,7 +66,6 @@
     "@_LOCALE_DROPDOWN": {},
     "english": "English",
     "russian": "Российский",
-    "ukrainian": "Українська",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:69]: Why deleted this string?

File: lib/presentation/pages/products_page/products_page.dart [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+          const SliverSizedBox(height: 100),
+          const SliverToBoxAdapter(child: _GradientSection()),
+          const SliverSizedBox(height: 100),
+          const SliverToBoxAdapter(child: Align(child: _ComponentsSection())),
+          const SliverSizedBox(height: 100),
+          const SliverToBoxAdapter(child: _PartnersSection()),
+          const SliverSizedBox(height: 100),
+          const SliverToBoxAdapter(child: Align(child: _FaqSection())),
+          const SliverSizedBox(height: 100),
+          const SliverToBoxAdapter(child: Footer()),
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/products_page.dart:63]: Pls delete all Align widgets and use alignment in sections

File: lib/presentation/pages/products_page/products_page.dart [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+class _Divider extends StatelessWidget {
+  const _Divider({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      constraints: const BoxConstraints(maxWidth: 1270),
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/products_page.dart:77]: I think you should replace Container width ConstrainedBox and replace padding by SliverSizedBox where divider is used

File: lib/presentation/pages/products_page/sections/components_section.dart [hunk 13]
```diff
@@ -0,0 +1,108 @@
+part of '../products_page.dart';
+
+class _ComponentsSection extends StatelessWidget {
+  const _ComponentsSection();
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: ConstrainedBox(
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/components_section.dart:10]: I think ConstrainedBox should be above padding

File: lib/presentation/pages/products_page/sections/components_section.dart [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+              textAlign: TextAlign.center,
+            ),
+            const SizedBox(height: 20),
+            Text(
+              context.localizations.mainProducts,
+              style: context.text.productsComponentItemDescription,
+              textAlign: TextAlign.center,
+            ),
+            const SizedBox(height: 40),
+            Align(
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/components_section.dart:26]: Why using Align withoud alignment property?

File: lib/presentation/pages/products_page/sections/components_section.dart [hunk 15]
```diff
@@ -0,0 +1,108 @@
+part of '../products_page.dart';
+
+class _ComponentsSection extends StatelessWidget {
+  const _ComponentsSection();
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/components_section.dart:4]: Add key pls

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+  const _FaqSection({Key? key}) : super(key: key);
+
+  List<Widget> _faqItems(BuildContext context) {
+    return [
+      Center(
+        child: Text(
+          "FAQ",
+          style: context.text.productsFaqSectionTitle,
+        ),
+      ),
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:13]: It shouldn't be in items. Move it and replace method by variable

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      constraints: const BoxConstraints(maxWidth: 820),
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: ListView.separated(
+        itemCount: _faqItems(context).length,
+        itemBuilder: (BuildContext context, int index) {
+          return _faqItems(context).elementAt(index);
+        },
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:47]: Replace by =>

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      constraints: const BoxConstraints(maxWidth: 820),
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: ListView.separated(
+        itemCount: _faqItems(context).length,
+        itemBuilder: (BuildContext context, int index) {
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:45]: Replace BuildContext by _, and delete type from index

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+          return _faqItems(context).elementAt(index);
+        },
+        separatorBuilder: (BuildContext context, int index) {
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:48]: Replace by _, __

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 20]
```diff
@@ -0,0 +1,166 @@
...
+
+class _FaqItemState extends State<_FaqItem>
+    with SingleTickerProviderStateMixin {
+  bool _isHovered = false;
+  bool _isExpanded = false;
+
+  static final Animatable<double> _easeInTween =
+      CurveTween(curve: Curves.easeIn);
+  static final Animatable<double> _halfTween =
+      Tween<double>(begin: 0.0, end: 0.25);
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:79]: Static variables should be above var variables

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 21]
```diff
@@ -0,0 +1,166 @@
...
+
+  late AnimationController _controller;
+  late Animation<double> _iconTurns;
+
+  Widget _buildRotatingIcon() {
+    return RotationTransition(
+      turns: _iconTurns,
+      child: Icon(Icons.arrow_forward, color: context.color.faqItemArrowFill),
+    );
+  }
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:89]: It should below build method

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 22]
```diff
@@ -0,0 +1,166 @@
...
+      } else {
+        _controller.reverse().then<void>((void value) {
+          if (!mounted) {
+            return;
+          }
+          setState(() {});
+        });
+      }
+    });
+  }
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:105]: Look bad

File: lib/presentation/pages/products_page/sections/faq_section.dart [hunk 23]
```diff
@@ -0,0 +1,166 @@
...
+                  const SizedBox(width: 10),
+                  _buildRotatingIcon(),
+                ],
+              ),
+              AnimatedContainer(
+                height: _isExpanded ? 100 : 0,
+                padding: const EdgeInsets.all(20),
+                duration: const Duration(milliseconds: 300),
+                child:
+                    Text(widget.answer, style: context.text.productsFaqAnswer),
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/faq_section.dart:158]: Add comma

File: lib/presentation/pages/products_page/sections/gradient_section.dart [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+    _InfoBlock(
+      title: context.localizations.partnersMultimetaNumber,
+      description: context.localizations.partnersMultimeta,
+    ),
+  ];
+}
+
+class _InfoBlock extends StatelessWidget {
+  final String title;
+  final String? description;
```
Comment (vovaklh) [inline lib/presentation/pages/products_page/sections/gradient_section.dart:47]: Set default value as empty string in constructor

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/l10n/app_ru.arb (hunk 10 above)
Comment (vitalii-petrun) [reply lib/l10n/app_ru.arb:69]: this string changed its locatiion after merge dev into this branch

File: lib/presentation/pages/products_page/sections/components_section.dart (hunk 14 above)
Comment (vitalii-petrun) [reply lib/presentation/pages/products_page/sections/components_section.dart:26]: it has Alignment.center as default value

**Review by vovaklh — APPROVED** [review]

**Pull request merged**